### DISCUSSION TITLE
How to set left margin for watermark / Floating Label to avoid colliding with InnerLeftContent ?

### DISCUSSION THREAD

**Discussion by CastelloBrancoTecnologia** [AvaloniaCommunity/Material.Avalonia#279] [Q&A]
![image](https://github.com/AvaloniaCommunity/Material.Avalonia/assets/138352584/984659d4-1266-42b5-8f1c-20dea7b17c1e)

inestead of this

![image](https://github.com/AvaloniaCommunity/Material.Avalonia/assets/138352584/9c6b42ba-9427-4552-b9b5-a12e29637814)

in 

   <TextBox  ..... >
                        <TextBox.InnerLeftContent>
                            <icons:MaterialIcon
                                Width="24"
                                Height="24"
                                Margin="4"
                                HorizontalAlignment="Left"
                                VerticalAlignment="Bottom"
                                Foreground="{DynamicResource PrimaryHueMidBrush}"
                                Kind="KeyVariant" />
                        </TextBox.InnerLeftContent>
                    </TextBox>

**Comment by SKProCH** [comment]
To be honest, its doesn't get a proper support now. You can try to edit Margin of watermark TextBlock via styles, but i think currently you should place icon leftside of TextBox. E.g:
```xaml
<StackPanel Orientation="Horizontal">
  <icons:MaterialIcon .../>
  <TextBox ...>
</StackPanel>
```

**Comment by SKProCH** [comment]
Progress can be tracked here: https://github.com/AvaloniaCommunity/Material.Avalonia/issues/280
Also, here is the rework progress: https://github.com/AvaloniaCommunity/Material.Avalonia/tree/textBoxRework
